### PR TITLE
Allow changing infiniteScrollDisabled

### PR DESCRIPTION
**Pull request by dylan-smith**
If infiniteScrollDisabled is changed, call setup() again to recreate the scroller.

Fix for Issue #195

### CODE REVIEW

File: src/modules/infinite-scroll.directive.ts [hunk 1]
```diff
@@ -43,8 +43,8 @@ export class InfiniteScrollDirective
     this.setup();
   }
 
-  ngOnChanges({ infiniteScrollContainer }: SimpleChanges) {
-    if (inputPropChanged(infiniteScrollContainer)) {
+  ngOnChanges({ infiniteScrollContainer, infiniteScrollDisabled }: SimpleChanges) {
+    if (inputPropChanged(infiniteScrollContainer) || inputPropChanged(infiniteScrollDisabled)) {
       this.destroyScroller();
```
Comment (orizens) [inline src/modules/infinite-scroll.directive.ts]: this should be handled differently - as the setup should occur only when infiniteScrollDisabled is false

Comment (dylan-smith) [reply src/modules/infinite-scroll.directive.ts]: I updated the PR to only call setup if its enabled, and to destroy the scroller when disabled.  Give a look, is that what you were thinking?

**Comment by ivandatio** [comment]
This works for me! I need this functionality for detecting changes on infiniteScrollDisabled, can this be merged asap? Good work!

**Comment by orizens** [comment]
this doesnt handle the disabled correctly

**Comment by dylan-smith** [comment]
@orizens not even with the update I did to the PR yesterday?  Can you elaborate on how you expect disabled to be handled?

**Comment by dylan-smith** [comment]
any input? happy to do the work, just need some info on what you're hoping for?

File: src/modules/infinite-scroll.directive.ts (hunk 1 above)
Comment (orizens) [reply src/modules/infinite-scroll.directive.ts]: thanks - i'll have a look and let you know.

File: src/modules/infinite-scroll.directive.ts [hunk 2]
```diff
@@ -40,13 +40,19 @@ export class InfiniteScrollDirective
...
-      this.setup();
+      
+      if ((!inputPropChanged(infiniteScrollDisabled) && !this.infiniteScrollDisabled) ||
```
Comment (orizens) [inline src/modules/infinite-scroll.directive.ts]: please cache and reuse -
```
const disabledChanged =  inputPropChanged(infiniteScrollDisabled);
```

Comment (orizens) [inline src/modules/infinite-scroll.directive.ts]: why you're using "this.infiniteScrollDisabled" with "this"?

Comment (dylan-smith) [reply src/modules/infinite-scroll.directive.ts]: the first check is against the function arg to check if ngOnChanges was called due to a change in the disabled property.  The second check is against the class-level prop to see if it was changed to false.

Comment (dylan-smith) [reply src/modules/infinite-scroll.directive.ts]: added caching

File: src/modules/infinite-scroll.directive.ts [hunk 3]
```diff
@@ -46,11 +46,14 @@ export class InfiniteScrollDirective
   }
 
   ngOnChanges({ infiniteScrollContainer, infiniteScrollDisabled }: SimpleChanges) {
-    if (inputPropChanged(infiniteScrollContainer) || inputPropChanged(infiniteScrollDisabled)) {
+    var containerChanged: boolean = inputPropChanged(infiniteScrollContainer);
```
Comment (orizens) [inline src/modules/infinite-scroll.directive.ts]: please use ```const``` for both vars.

Comment (dylan-smith) [reply src/modules/infinite-scroll.directive.ts]: made both changes, rebased onto updated master.

**Pull request merged**